### PR TITLE
mon/ConfigMonitor: Show localized name in "config dump --format json" output

### PR DESCRIPTION
**Pull request by sseshasa**
The "ceph config dump" command without the json formatted output shows the
localized option names and their values. An example of a normalized vs localized
option is shown below:

Normalized: mgr/dashboard/ssl_server_port (maintaned within Option struct)
Localized: mgr/dashboard/x/ssl_server_port (maintained in mon store)

But the "ceph config dump --format json*" output showed the normalized option
names which was not consistent with the "config dump" output. The output of
the command along with variations for pretty printing must show the same content.

This commit introduces a new member within the ConfigMap's MaskedOption struct
called "localized_name". This is initialized to the localized name as part of
ConfigMonitor::load_config() method.

The MaskedOption::dump() used for the json formatting is modified to display the
localized_name instead of the normalized name.

Fixes: https://tracker.ceph.com/issues/62379
Signed-off-by: Sridhar Seshasayee <sseshasa@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by sseshasa** [comment]
Test Results With Fix:


```
$ ./bin/ceph config dump | grep server_
mgr           advanced  mgr/dashboard/x/ssl_server_port   41728         * 
mgr           advanced  mgr/prometheus/x/server_port      9283            
mgr           advanced  mgr/restful/x/server_port         42728         *

$ ./bin/ceph config dump --format json | jq | grep -A 5 -B 3 server_
  },
  {
    "section": "mgr",
    "name": "mgr/dashboard/x/ssl_server_port",
    "value": "41728",
    "level": "advanced",
    "can_update_at_runtime": false,
    "mask": ""
  },
  {
    "section": "mgr",
    "name": "mgr/prometheus/x/server_port",
    "value": "9283",
    "level": "advanced",
    "can_update_at_runtime": true,
    "mask": ""
  },
  {
    "section": "mgr",
    "name": "mgr/restful/x/server_port",
    "value": "42728",
    "level": "advanced",
    "can_update_at_runtime": false,
    "mask": ""
  },

```

**Comment by sseshasa** [comment]
@rzarzynski @neha-ojha PTAL when you can.

**Review by rzarzynski — APPROVED** [review]
Approving for `main` while being a bit nervous about fallouts / users relying on the buggy behavior. Might be worth a CDM.

**Comment by sseshasa** [comment]
> Approving for `main` while being a bit nervous about fallouts / users relying on the buggy behavior. Might be worth a CDM.

@rzarzynski I posted changes pertaining to this PR on the ceph-users mail list (https://www.spinics.net/lists/ceph-users/msg78499.html) and didn't get any response so far. Therefore, I think this change shouldn't cause a major fallout. We can also bring this up in other dev or community meetups to be sure.

**Comment by rzarzynski** [comment]
@neha-ojha: points out we should document this in release notes.

**Comment by rzarzynski** [comment]
@yuriw, @ljflores: ping.

**Comment by ljflores** [comment]
It's in testing now @rzarzynski !

**Comment by sseshasa** [comment]
> @neha-ojha: points out we should document this in release notes.

@rzarzynski Sure, I will add to the release notes.

**Comment by sseshasa** [comment]
> > @neha-ojha: points out we should document this in release notes.
> 
> @rzarzynski Sure, I will add to the release notes.

@rzarzynski @neha-ojha Added the release note. Please verify.

**Comment by sseshasa** [comment]
jenkins test api

File: PendingReleaseNotes [hunk 1]
```diff
@@ -198,6 +198,8 @@
   also change read/write permissions in a capability that the entity already
   holds. If the capability passed by user is same as one of the capabilities
   that the entity already holds, idempotency is maintained.
+* `ceph config dump --format <json|xml>` will now show the localized option
```
Comment (rzarzynski) [inline PendingReleaseNotes:201]: nit: _shows now_?

Comment (sseshasa) [reply PendingReleaseNotes]: @rzarzynski I modified the note to include bit more detail. PTAL.

**Comment by pdvian** [comment]
https://pulpito.ceph.com/lflores-2023-09-06_18:19:12-rados-wip-yuri6-testing-2023-08-28-1308-distro-default-smithi/

Failures, unrelated:
1. https://tracker.ceph.com/issues/59142
2. https://tracker.ceph.com/issues/59196
3. https://tracker.ceph.com/issues/55347 
4. https://tracker.ceph.com/issues/62084
5. https://tracker.ceph.com/issues/62975                 <<<---- New Tracker
6. https://tracker.ceph.com/issues/62449
7. https://tracker.ceph.com/issues/61774
8. https://tracker.ceph.com/issues/53345

Details:
1. mgr/dashboard: fix e2e for dashboard v3 - Ceph - Mgr - Dashboard
2. ceph_test_lazy_omap_stats segfault while waiting for active+clean - Ceph - RADOS
3. SELinux Denials during cephadm/workunits/test_cephadm
4. task/test_nfs: AttributeError: 'TestNFS' object has no attribute 'run_ceph_cmd' - Ceph - CephFS
5. site-packages/paramiko/channel.py: OSError: Socket is closed - Infrastructure
6. test/cls_2pc_queue: TestCls2PCQueue.MultiProducer and TestCls2PCQueue.AsyncConsumer failure - Ceph - RGW
7. centos 9 testing reveals rocksdb "Leak_StillReachable" memory leak in mons - Ceph - RADOS
8. Test failure: test_daemon_restart (tasks.cephadm_cases.test_cli.TestCephadmCLI) - Ceph - Orchestrator

**Pull request merged**